src/main/java/net/sf/jabref/logic/importer/fetcher/MrDLibFetcher.java

```diff
@@ -0,0 +1,145 @@
+/**
+ *
+ */
+package net.sf.jabref.logic.importer.fetcher;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+import javax.net.ssl.SSLContext;
+
+import net.sf.jabref.logic.importer.EntryBasedFetcher;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.ParserResult;
+import net.sf.jabref.logic.importer.fileformat.MrDLibImporter;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.BuildInfo;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContexts;
+
+/**
+ *  This class is responible to get the recommendations from MDL
+ *
+ */
+public class MrDLibFetcher implements EntryBasedFetcher {
+
+    private static final String NAME = "MDL_FETCHER";
+    private static final Log LOGGER = LogFactory.getLog(MrDLibFetcher.class);
+    private final String LANGUAGE;
+    private final String VERSION;
+    private final JabRefPreferences prefs = JabRefPreferences.getInstance();
+
+    public MrDLibFetcher() {
+        LANGUAGE = prefs.get(JabRefPreferences.LANGUAGE);
+        VERSION = new BuildInfo().getVersion().getFullVersion();
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) throws FetcherException {
+        String response = makeServerRequest(entry.getLatexFreeField("title").get());
+        MrDLibImporter importer = new MrDLibImporter();
+        ParserResult parserResult = new ParserResult();
+        try {
+            if (importer.isRecognizedFormat(new BufferedReader(new StringReader(response)))) {
+                parserResult = importer.importDatabase(new BufferedReader(new StringReader(response)));
+            } else {
+                // For displaying An ErrorMessage
+                BibEntry errorBibEntry = new BibEntry();
```

Comment:
I think that is a nice way to handle the error message. It is stored within a bibentry. In both cases, with or without error, a bib entry is displayed. Exception handling at this point seems to be more inconsistent and reduces readability in the related article tab class.